### PR TITLE
Handle Exception in seekToLastCheckpoint in AbstractKafkaMirrorMakerConnectorTask

### PR DESCRIPTION
**Pull request by vmaheshw**
In the case when the connector task is stopped and connector thread is not blocked on the poll call, the next consumer call gets WakeUpException, which in our fail case is "seekToLastCheckpoint" call. This results in not attempting to commitCheckpoint before the task dies. In case of FlushlessEventProducer, it is still safe to commitSafeOffsets, to minimize the duplication of records.

### CODE REVIEW

**Review by somandal — APPROVED** [review]
sorry, forgot to approve this.

File: datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/mirrormaker/KafkaMirrorMakerConnectorTask.java [hunk 1]
```diff
@@ -485,6 +483,30 @@ public KafkaDatastreamStatesResponse getKafkaDatastreamStatesResponse() {
         _isFlushlessModeEnabled ? _flushlessProducer.getInFlightMessagesCounts() : Collections.emptyMap());
   }
 
+  @VisibleForTesting
+  protected void seekToLastCheckpoint(Set<TopicPartition> topicPartitions) {
+    try {
+      super.seekToLastCheckpoint(topicPartitions);
+    } catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (vmaheshw) [reply datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/mirrormaker/KafkaMirrorMakerConnectorTask.java:490]: The reason to catch any exception is, it does not really matter. safeOffset should be good to commit irrespective of the exception type.

File: datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java [hunk 2]
```diff
@@ -274,7 +274,6 @@ protected void rewindAndPausePartitionOnException(TopicPartition srcTopicPartiti
       // Seek to last checkpoint failed. Throw an exception to avoid any data loss scenarios where the consumed
       // offset can be committed even though the send for that offset has failed.
       String errorMessage = String.format("Partition rewind for %s failed due to ", srcTopicPartition);
-      _logger.error(errorMessage, e);
```
[the remark this answers is not in the document]
Comment (vmaheshw) [reply datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java:277]: This error was duplicate in the logs. The caller of this method prints the exception, or there is a print when the thread dies with the exception.

**Review by somandal — APPROVED** [review]

**Review by jzakaryan — APPROVED** [review]
LGTM

**Pull request merged**